src/getMessageAuthor.ts

```diff
@@ -1,21 +1,36 @@
 import {endGroup, info, isDebug, startGroup, warning} from '@actions/core'
 import {context} from '@actions/github'
-import {OctokitClient, User} from './github/types'
-import {senderFromPayload} from './github/webhook'
+import type {Commit} from '@octokit/webhooks-types'
+import {OctokitClient} from './github/types'
+import {GitHubSender, isPushEvent, senderFromPayload} from './github/webhook'
 import {SlackClient} from './slack/SlackClient'
 import {MemberWithProfile, MessageAuthor} from './slack/types'
 
+export const GH_MERGE_QUEUE_BOT_USERNAME = 'github-merge-queue[bot]'
+
 export async function getMessageAuthor(
   octokit: OctokitClient,
   slack: SlackClient
 ): Promise<MessageAuthor | null> {
   startGroup('Getting message author')
 
+  const githubSender = await getGitHubSender(octokit)
```

Comment:
@Blacksmoke16, a23484b5c218c0ec024e015ae8e464d463d4c7d7 takes one more pass at this after realizing the subtle (but important) difference between the GitHub "sender" and "user". What are your thoughts on these changes?

Unit tests still pass which is good, but can we test this again E2E? Happy to hop on a call to chat live as well.